test/js/api/domain/post/files-spec.js

```diff
@@ -0,0 +1,239 @@
+        }
+    };    
+    
+    var getFiles = function(filenames) {
+        filenames = filenames || [];
+        
+        var files = [];
+        
+        _.each(filenames, function(filename) {
+            files.push({'name':filename, 'file':filename});
+        });
+        
+        return files;
+    }
+    
+    var respond = function(req, res, options) {
+        files.__set__('globalConfig', options);                
+        files(req, res, options);            
+    };    
+            
+    it('should populate config for the default scenario', function() {
+        
+        var res = httpRes();
+        
+        var config = {
+            app: {data: defaultData},
+            api:{
+                defaultApi: {
+                    source: defaultSource
+                }
+            }
+        };
+                
+        respond(httpReq({api: defaultApi}), res, config);
+        assert.deepEqual(res._getData(), {  
+           scenarios:[  
+              'default'
+           ],
+           config:{  
+              api:'defaultApi',
+              name:'test_req',
+              file:'request_0'
+           },
+           default:[  
+              {  
+                 name:'a',
+                 file:'a'
+              },
+              {  
+                 name:'b',
+                 file:'b'
+              },
+              {  
+                 name:'c',
+                 file:'c'
+              }
+           ]
+        });        
+    });
+
+    it('should populate config for the provided scenario', function() {
+                
+        var res = httpRes();
+        
+        var config = {
+            app: {data: debugData},
+            api:{
+                debugApi: {
+                    source: debugSource,
+                    scenarios: {debugScenario: {source: debugSource}}
+                }
+            }
+        };
+                
+        respond(httpReq({api: debugApi}), res, config);        
+        assert.deepEqual(res._getData(), {  
+           scenarios:[  
+              'default',
+              'debugScenario'
+           ],
+           config:{  
+              api:'debugApi',
+              name:'test_req',
+              file:'request_0'
+           },
+           default:[  
+              {  
+                 name:'d',
+                 file:'d'
+              },
+              {  
+                 name:'e',
+                 file:'e'
+              },
+              {  
+                 name:'f',
+                 file:'f'
+              }
+           ],
+           debugScenario:[  
+              {  
+                 name:'d',
+                 file:'d'
+              },
+              {  
+                 name:'e',
+                 file:'e'
+              },
+              {  
+                 name:'f',
+                 file:'f'
+              }
+           ]
+        });           
+    });
+    
+    it('should skip an active scenario', function() {
```

Comment:
It doesn't "skip" an active scenario, "active" is not used here. It holds what scenario the "proxy/simple" / "proxy/params" module should use when matching the request